lib/v2/bilibili/cache.js

```diff
@@ -1,27 +1,75 @@
 const got = require('@/utils/got');
+const UA = require('@/utils/rand-user-agent')({ browser: 'chrome', os: 'windows', device: 'desktop' });
 
 module.exports = {
-    getUsernameFromUID: (ctx, uid) => {
-        const key = 'bili-username-from-uid-' + uid;
-        return ctx.cache.tryGet(key, async () => {
-            const { data: nameResponse } = await got(`https://api.bilibili.com/x/space/acc/info?mid=${uid}&jsonp=jsonp`, {
+    getCookie: async (ctx) => {
+        const key = 'cookie';
+        let cookie = await ctx.cache.get(key);
+        if (!cookie) {
+            const response = await got({
+                method: 'get',
+                url: `https://www.bilibili.com/`,
                 headers: {
-                    Referer: `https://space.bilibili.com/${uid}`,
+                    'User-Agent': UA
                 },
             });
-            return nameResponse.data.name;
-        });
+            cookie = response.headers['set-cookie'];
+            ctx.cache.set(key, cookie);
+        }
+        return cookie;
+    },
+    getUsernameFromUID: async (ctx, uid) => {
+        const cookie = await module.exports.getCookie(ctx);
+        const key = 'bili-username-from-uid-' + uid;
+        let name = await ctx.cache.get(key);
+        if (!name) {
+            await got({
+                method: 'get',
+                url: `https://space.bilibili.com/${uid}/`,
```

Comment:
got request can be simplified
```suggestion
            await got(`https://space.bilibili.com/${uid}/`, {
```